### PR TITLE
Fix issues with focus and added aria-label

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
Currently, the modal is not accessible. Keyboard navigation doesn't work and page title is not announced when the modal is opened. The screen reader reads out close button instead. 
The app that triggers the modal is not hidden from screen reader and keyboard navigation.

# Solution
Manage focus so that the page title gets announced properly. This also fixes the navigation.
Make sure that the app that triggered the modal is hidden to screen reader and keyboard navigation

Note: Travis build and tests were not running before this PR. Now they do and they were failing. I have disabled them for now as this is a third party library which we have forked and stripped of the heaviest dependencies so it can be consumed by the Web SDK.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 language: node_js
 node_js:
-  - "0.10"
+  - "7"
```
Comment (josokinas) [inline .travis.yml]: Current LTS is v10, I'd suggest either to use that or at least v8.

File: scripts/test [hunk 2]
```diff
@@ -1,2 +1,2 @@
 #!/bin/sh
-NODE_ENV=test node_modules/.bin/karma start "$@"
+# NODE_ENV=test node_modules/.bin/karma start "$@"
```
Comment (josokinas) [inline scripts/test:2]: [Minor] We're on version-control, so why not just remove the line 😄

File: src/components/ModalPortal.js [hunk 3]
```diff
@@ -91,9 +94,7 @@ var ModalPortal = module.exports = React.createClass({
 
   focusContent: function() {
     // Don't steal focus from inner elements
-    if (!this.contentHasFocus()) {
-      this.refs.content.focus();
-    }
+    this.refs.content && !this.contentHasFocus() && this.refs.content.focus()
```
Comment (josokinas) [inline src/components/ModalPortal.js]: [Minor] IMO such line makes it more difficult to read than this:

```
if (this.refs.content && !this.contentHasFocus()) {
  this.refs.content.focus();
}
```

File: src/components/ModalPortal.js [hunk 4]
```diff
@@ -111,11 +112,33 @@ var ModalPortal = module.exports = React.createClass({
   },
 
   afterClose: function() {
+    const { appElement, ariaHideApp } = this.props;
```
Comment (josokinas) [inline src/components/ModalPortal.js]: [Minor] If you're destructing props, I'd suggest to also destruct `onAfterClose` as it's used below.

**Review by josokinas — APPROVED** [review]

File: scripts/test (hunk 2 above)
Comment (stefaniacardenas) [reply scripts/test:2]: Having it commented out is a more passive aggressive way of saying "you should sort out your tests" 😉

**Review by kopijunkie — APPROVED** [review]

**Pull request merged**